File: charCount/js/charCount.js
exports.charCount = function(str) {
    const charObj = {};
    charArr = [];

    for (let char of str){
        if (char === ' ') {
            continue;
        } else if (charObj[char]) {
            charObj[char]++;
        } else {
            charObj[char] = 1;
        }
    }

    for ([key, value] of Object.entries(charObj)) {
        charArr.push([key, value]);
    }
    
    charArr.sort((a,b) => b[1] - a[1]);
    
    for (let subList = 0; subList < charArr.length; subList++) {
        for (let idx = 0; idx < charArr.length - 1; idx++) {
            if (charArr[idx][1] === charArr[idx+1][1]) {
                if (charArr[idx][0] > charArr[idx+1][0]) {
                    let temp = charArr[idx];
                    charArr[idx] = charArr[idx+1];
                    charArr[idx+1] = temp;
                }
            }
        }
    }
    return charArr


};

File: charCount/python/char_count.py
def char_count(str):
  char_dict = {}
  char_list = []

  for char in str:
    if char == ' ':
      continue
    elif char in char_dict:
      char_dict[char] += 1
    else:
      char_dict[char] = 1
 
  for key, value in char_dict.items():
    char_list.append([key, value])

  char_list.sort(key=lambda x: x[1], reverse=True)

  for each_list in range(len(char_list)-1):
    for item in range(len(char_list)-1):
      if char_list[item][1] == char_list[item+1][1]:
        if char_list[item][0] > char_list[item+1][0]:
          temp = char_list[item]
          char_list[item] = char_list[item+1]
          char_list[item+1] = temp

  return char_list